### PR TITLE
FileResource implementation

### PR DESCRIPTION
**Pull request by Alex009**
- `files` directory in `MR` for files
- on Android used raw resources for storage of files
- on iOS all files contains in `files` directory of bundle

# OS differences
if we use same filename in different modules - on android will be used one of this files (from parent module), on iOS will be used different files.
if we have:
- `mpp-library/src/commonMain/resources/MR/files/test.txt` with `first` contend
- `mpp-library/nested/src/commonMain/resources/MR/files/test.txt` with `second` content
on android if we read `mpplib.MR.files.test` we got `first` and for `mpplib.nested.MR.files.test` too will be `first`
but on iOS will be as expected: `mpplib.MR.files.test` = `first`, `mpplib.nested.MR.files.test` = `second`

how it solve i don't know now

### CODE REVIEW

File: gradle-plugin/src/main/kotlin/dev/icerock/gradle/generator/android/AndroidFilesGenerator.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    override fun generateResources(
+        resourcesGenerationDir: File,
+        files: List<FileSpec>
+    ) {
+        val targetDir = File(resourcesGenerationDir, "raw")
+        targetDir.mkdirs()
+
+        files.forEach { (key, file) ->
+            val fileName = keyToResourceId(key) + "." + file.extension
+            file.copyTo(File(targetDir, fileName))
```
Comment (Tetraquark) [inline gradle-plugin/src/main/kotlin/dev/icerock/gradle/generator/android/AndroidFilesGenerator.kt:42]: There may be an `IOException`. Is it necessary to handle it here?

File: gradle-plugin/src/main/kotlin/dev/icerock/gradle/generator/ios/IosFilesGenerator.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    override fun generateResources(
+        resourcesGenerationDir: File,
+        files: List<FileSpec>
+    ) {
+        val targetDir = File(resourcesGenerationDir, "files")
+        targetDir.mkdirs()
+
+        files.forEach { (_, file) ->
+            file.copyTo(File(targetDir, file.name))
```
Comment (Tetraquark) [inline gradle-plugin/src/main/kotlin/dev/icerock/gradle/generator/ios/IosFilesGenerator.kt:38]: Same with `IOException`.

File: sample/android-app/src/main/java/com/icerockdev/MainActivity.kt [hunk 3]
```diff
@@ -33,6 +34,12 @@ class MainActivity : AppCompatActivity() {
 
         stringDescTextView.text = Testing.getStringDesc().toString(context = this)
         stringDescTextView.typeface = Testing.getFont2().getTypeface(context = this)
+
+        listOf(
+            Testing.getTextFile().readText(context = this),
+            Testing.getJsonFile().readText(context = this),
+            Testing.getNestedJsonFile().readText(context = this)
```
Comment (Tetraquark) [inline sample/android-app/src/main/java/com/icerockdev/MainActivity.kt:41]: Is need to handle exceptions here when working with files?

Comment (Alex009) [reply sample/android-app/src/main/java/com/icerockdev/MainActivity.kt:41]: this files is contained in resources and plugin generate safe code for access to this resources, so i not see any exceptions what really can be throw

File: gradle-plugin/src/main/kotlin/dev/icerock/gradle/generator/android/AndroidFilesGenerator.kt (hunk 1 above)
Comment (Alex009) [reply gradle-plugin/src/main/kotlin/dev/icerock/gradle/generator/android/AndroidFilesGenerator.kt:42]: in gradle generator - no. if something will be wrong this code should crash and gradle execution will be failed

File: gradle-plugin/src/main/kotlin/dev/icerock/gradle/generator/ios/IosFilesGenerator.kt (hunk 2 above)
Comment (Alex009) [reply gradle-plugin/src/main/kotlin/dev/icerock/gradle/generator/ios/IosFilesGenerator.kt:38]: in gradle generator - no. if something will be wrong this code should crash and gradle execution will be failed

File: resources/src/androidMain/kotlin/dev/icerock/moko/resources/FileResource.kt [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+actual class FileResource(
+    @RawRes
+    val rawResId: Int
+) {
+    fun readText(context: Context): String {
+        val resources: Resources = context.resources
+        val inputStream: InputStream = resources.openRawResource(rawResId)
+        val bufferedReader = BufferedReader(InputStreamReader(inputStream))
+        return bufferedReader.readText()
```
[the remark this answers is not in the document]
Comment (Alex009) [reply resources/src/androidMain/kotlin/dev/icerock/moko/resources/FileResource.kt:22]: yes, thanks! i will fix it

**Pull request merged**